vmm/src/memory_manager.rs

```diff
@@ -228,8 +241,30 @@ impl MemoryManager {
             mem_end.unchecked_add(1)
         };
 
+        let mut virtiomem_region = None;
+        let mut virtiomem_resize = None;
         if let Some(size) = hotplug_size {
-            start_of_device_area = start_of_device_area.unchecked_add(size);
+            if hotplug_method == HotplugMethod::VirtioMem {
+                let block_size = vm_virtio::VIRTIO_MEM_DEFAULT_BLOCK_SIZE as u64;
+                let start_addr = if start_of_device_area < arch::layout::MEM_32BIT_RESERVED_START {
+                    arch::layout::RAM_64BIT_START
+                } else {
+                    GuestAddress(
+                        (start_of_device_area.0 + block_size - 1) / block_size * block_size,
+                    )
+                };
```

Comment:
Same thing here, do we need the `MemoryManager` to know about the details, by relying on `VIRTIO_MEM_DEFAULT_BLOCK_SIZE`?

Reply:
Yes, the start address of virtio-mem must be aligned with VIRTIO_MEM_DEFAULT_BLOCK_SIZE.
This address is allocated in MemoryManager.
So it need know VIRTIO_MEM_DEFAULT_BLOCK_SIZE.